### ISSUE TITLE
1.2.0 Release

### ISSUE THREAD

**Issue by abn** [python-poetry/poetry#5586]
Trying to gather the list of actual blockers for 1.2.0 final release. If there is a PR open that absolutely must be included in the 1.2.0 release, please link them here. Issues, unless they are deemed critical (ie. regression that fundamentally breaks workflows), are unlikely to be considered at this stage.

A link and a short note on why you think is important would suffice. If you agree that a linked issue/pr in the comments is important use :+1:  else use :-1:  to vote.

Let's avoid bike shedding in here so that the maintainers can focus on the signal.

# Blockers
These are a subset of the [1.2 Milestone](https://github.com/python-poetry/poetry/milestone/5).

- https://github.com/python-poetry/poetry-core/pull/348
- https://github.com/python-poetry/poetry/pull/5673
- https://github.com/python-poetry/poetry/pull/5450
- https://github.com/python-poetry/poetry/pull/5581
- https://github.com/python-poetry/poetry/issues/4242
- https://github.com/python-poetry/poetry-plugin-export/pull/56
- https://github.com/python-poetry/poetry/pull/5587
- https://github.com/python-poetry/poetry/pull/5590
- https://github.com/python-poetry/poetry/pull/6234
- https://github.com/python-poetry/poetry/issues/5593
- https://github.com/python-poetry/poetry/issues/5633
- https://github.com/python-poetry/poetry/issues/5675
- https://github.com/python-poetry/poetry/issues/5682
- https://github.com/python-poetry/poetry/issues/4718
- https://github.com/python-poetry/poetry/issues/5715
- https://github.com/python-poetry/poetry/issues/5727
- https://github.com/python-poetry/poetry/pull/5725
- https://github.com/python-poetry/poetry/issues/5331
- https://github.com/python-poetry/poetry/issues/6054
- https://github.com/python-poetry/poetry/issues/6184

# Nice to haves
- https://github.com/python-poetry/poetry/pull/5811
- https://github.com/python-poetry/poetry/pull/5172
- https://github.com/python-poetry/poetry/pull/5451
- https://github.com/python-poetry/poetry/pull/5442
- https://github.com/python-poetry/poetry/pull/5401
- https://github.com/python-poetry/poetry/pull/5385
- https://github.com/python-poetry/poetry/pull/5718
- https://github.com/python-poetry/poetry/pull/5841
- https://github.com/python-poetry/poetry/pull/5935

# General Notes
## PEP 610 Changes

Due to changes required in #5362, when upgrading from Poetry `1.1` releases, existings environments might see the following "no update" logs. This is because, with `1.2.0a1`, we fixed an issue with the executor that wrote incorrect PEP 610 metadata.

```console
  • Updating tox (3.25.0 /home/abn/.cache/pypoetry/artifacts/a2/33/48/b1983b149095eaa5fbb5f0d48841785f6778b0b3fcb8be53a7506b349a/tox-3.25.0-py2.py3-none-any.whl -> 3.25.0)
  • Updating types-requests (2.27.29 /home/abn/.cache/pypoetry/artifacts/bd/20/0d/db608c01e91c5a34cdf7f8ac0407dde334f95d16c83985b870b17f508f/types_requests-2.27.29-py3-none-any.whl -> 2.27.29)
```

**Comment by neersighted** [comment]
I would like to get #5479 or a successor in -- I have yet to look closely to validate the approach, but I think a 1.2 release should solve the underlying issue (#4678), and the root cause shouldn't be too gory.

**Comment by matteosantama** [comment]
#4979 was deemed a duplicate and closed, but it is different in that it represents a 1.1.x -> 1.2.x regression. As far as I can tell, build scripts will be broken by 1.2.0

**Comment by Secrus** [comment]
I'd say this: #5529, as big update like this, is good moment to remove 2.7 support. Still working on it though.

**Comment by mkniewallner** [comment]
https://github.com/python-poetry/poetry/pull/3767 may be really nice to have too.

**Comment by fsgnil** [comment]
Hey folks,

My company is very interested in using poetry as part of our python infrastructure. We have been trying poetry internally and feeling really good about it. The problem is that we need to use poetry plugin to support some of our internal pypi set up. So I'm wondering if there is an estimated release date for poetry 1.2.0.

Thanks!

**Comment by neersighted** [comment]
> Hey folks,
> 
> My company is very interested in using poetry as part of our python infrastructure. We have been trying poetry internally and feeling really good about it. The problem is that we need to use poetry plugin to support some of our internal pypi set up. So I'm wondering if there is an estimated release date for poetry 1.2.0.
> 
> Thanks!

As with most solely voulenteer-driven projects, the answer is unfortunately when it's ready. The 1.2 release is quite big due to our poor release cadence, so this is taking a decent amount of time to cook. We hope to have a RC out in a week or two, and have a final release a week after that. Setbacks like major regressions or complications in the final few blockers may affect that timeline.

If you are interested in helping us get over the finish line, we're always looking for additional help. I'd suggest joining our Discord server and asking if there's anything you can help with in #contributing -- in particular, we're still looking for more volunteers for triage, as well as product/release management.

**Comment by fsgnil** [comment]
> 

Thanks for the quick reply.

I'll see if I can contribute too.

**Comment by yodaldevoid** [comment]
As `get-poetry.py` is getting depreciated with 1.2.0, I'd ask that #5331 be added as a blocker as currently some people are unable to install poetry on Windows with the new installer.

**Comment by Darsstar** [comment]
I would like to see #5727 fixed. I posted working code in the issue, so now I get to dive in the test cases for solving in order to be a good open source citizen and provide a failing test case.

**Comment by neersighted** [comment]
> As `get-poetry.py` is getting depreciated with 1.2.0, I'd ask that #5331 be added as a blocker as currently some people are unable to install poetry on Windows with the new installer.

See my comments on that issue -- I don't think we're going to call this a blocker, but I would like to see MS Store Python working before we ship 1.2 if it's not going to hold up the whole release. The issue likely lies in the installer which means it can be fixed after release with no additional patch, anyway.

**Comment by yodaldevoid** [comment]
> > As `get-poetry.py` is getting depreciated with 1.2.0, I'd ask that #5331 be added as a blocker as currently some people are unable to install poetry on Windows with the new installer.
> 
> See my comments on that issue -- I don't think we're going to call this a blocker, but I would like to see MS Store Python working before we ship 1.2 if it's not going to hold up the whole release. The issue likely lies in the installer which means it can be fixed after release with no additional patch, anyway.

Fair enough. Thank you for taking a look.

**Comment by ashnair1** [comment]
Could #5811 be added in the nice to haves?

**Comment by fredrikaverpil** [comment]
@abn I would like to ad https://github.com/python-poetry/poetry/pull/5935 to this list as 1.1.13 is working fine but without this change, 1.2 is unusable in private orgs.

**Comment by dimbleby** [comment]
Some desirable fixes that are currently on the backlog below.

Having said which: I would personally be happy to see 1.2.0 cut right now, if we can improve the release cadence - things that don't make 1.2.0 can always go into 1.2.x 

If there are issues that would help to make the release process faster / easier / more regular: suggest that advertising them prominently in this repository so that folk can chip in and help would be a good move.

Anyway these I think just want someone to hit the merge button
- #5871 
- https://github.com/python-poetry/cleo/pull/165
- https://github.com/python-poetry/cleo/pull/166

This needs a poetry-core release but looks like a good correctness fix
- #5770

And I have a couple of fixes relating to the correctness of `export` which require not only a poetry-core release but also some sort of coordination (because CI mess) with poetry-plugin-export): 
- #5711 
- #5819

**Comment by neersighted** [comment]
> As someone who interacts with Python novices a lot and introduces them to Poetry, I would ask for the following to be considered for 1.2:
> 
>     * [Why does new `install.python-poetry.org` script not add `poetry` to `PATH`? #5917](https://github.com/python-poetry/poetry/discussions/5917)
> 
>     * [[Doc] Add PowerShell snippet in the installation instructions (web & scripts) to allow Windows users to modify `%PATH%` easier #5915](https://github.com/python-poetry/poetry/issues/5915)
> 
> 
> I think that once the new installer becomes default on the website, most users will start using it. The new installer does not add `poetry` to the `PATH` automatically. While this might sound trivial to those of us experienced, for novice Windows users this could be a big enough paper cut that they don't continue trying out Poetry (I have witnessed this kind of thing first hand).
> 
> Two paths forward I see:
> 
>     * If it's intentional that Poetry does not want to support adding `poetry` to the `PATH` automatically, consider [[Doc] Add PowerShell snippet in the installation instructions (web & scripts) to allow Windows users to modify `%PATH%` easier #5915](https://github.com/python-poetry/poetry/issues/5915) or similar as a minimal step to help new users quickly get the `PATH` fixed up
> 
>     * Restore support to the new installer for adding `poetry` to the `PATH` automatically to smooth onboarding new users

See my answer at your linked discussion question -- in short, I don't mind adding a snippet, though I think a 'copy and paste' Powershell snippet is somewhat problematic as not all Windows users use Powershell. In general, as Poetry is a CLI program we do expect users to have enough wherewithall to Google how to modify their PATH -- on Windows it's a well-documented/known GUI.

**Comment by neersighted** [comment]
New regression/blocker for 1.2: https://github.com/python-poetry/poetry/issues/6054

**Comment by dimbleby** [comment]
All the blockers are closed.  Ship it!

Really: is there anything blocking the release, or is it "just" (yes I know) a question of maintainer time?

- If there is some blocking issue or issues, please share, perhaps someone will want to take it / them on
- If the bottleneck is maintainer time: is there anything the rest of us can do to help?

**Comment by neersighted** [comment]
Added a new blocker that was just discovered -- there is already a PR pending, we are just waiting on tests.

**Comment by Kobold** [comment]
You folks are awesome. Thank you for the update @neersighted. I can't say how excited I am for the 1.2 release.

**Comment by strangemonad** [comment]
I've been running 1.2.0b3. is there anything else we can do to help test for the release?

**Comment by radoering** [comment]
1.2.0rc1 is just around the corner (#6175). Testing it for regressions is welcome. If there are no major issues, the release should follow shortly after.

**Comment by neersighted** [comment]
1.2.0rc1 is tagged and is in the stabilization phase. The 1.2 branch is created and will be accepting backports for any last-minute issues found with the release candidate. Assuming we don't have any regressions, 1.2.0 will be released next Monday.

**Comment by rjurney** [comment]
1.2 will rock your socks. It will revolutionize the Python world!

**Comment by neersighted** [comment]
A quick status update for those following this issue: We had to release 1.2.0rc2 with some important fixes plus the reintroduction of a removed plugin API. We are shooting for this Wednesday for the final release of 1.2.

The related PRs are:

* https://github.com/python-poetry/website/pull/68 (blog)
* https://github.com/python-poetry/poetry-core/pull/455 (poetry-core)
* https://github.com/python-poetry/poetry/pull/6260 (Poetry)

**Comment by neersighted** [comment]
> We are unlikely to be able to support poetry 1.2.0 on conda-forge unless #5980 gets figured out. conda fundamentally cannot handle circular dependencies.

This is known and we do not plan to address it as part of 1.2. We have discussed a path for breaking the circular dependency in a way that is compatible with conda-forge and a pleasant user experience, which will start in 1.3 and likely be finalized in 1.4.

Please don't let the version numbers scare you, the intention is to release early and often and to minimize the number of backports to help keep project velocity up. 1.2 is something of an aberration but we have a revitalized core team and much more momentum (plus, no pandemic disrupting people's lives).

**Comment by xylar** [comment]
@neersighted, okay, thanks.  We're already running into packages that can't be built on conda-forge because of not supporting 1.2.0 prereleases so it's good to know that this problem hopefully won't be long lived.

**Comment by BradenM** [comment]
:rocket: Thank you guys for the hard work!
